### PR TITLE
Added an example to send data to server

### PR DESCRIPTION
**Pull request by hemanthrajv**
Added an example that demonstrates sending data to server using a `POST` request. 

Added with respect to the request: #3679

### CODE REVIEW

File: src/docs/cookbook/networking/send-data.md [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+  2. Send data to server using the `http` package.
+  3. Convert the response into a custom Dart object.
+  4. Getting `title` from user input
+  5. Display the response on screen.
+
+## 1. Add the `http` package
+
+To install the `http` package, add it to the dependencies section
+of the `pubspec.yaml`. You can find the latest version of the
+[http package][] the pub.dev.
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: the pub.dev => on pub.dev.

File: src/docs/cookbook/networking/send-data.md [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+```
+
+### Convert the `http.Response` to a `Album`
+
+Now, use the following steps to update the `createAlbum()`
+function to return a `Future<Album>`:
+
+  1. Convert the response body into a JSON `Map` with the `dart:convert`
+     package.
+  2. If the server does return an CREATED response with a status code of 201, then convert
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: does return => returns
an CREATED => a `CREATED`

File: src/docs/cookbook/networking/send-data.md [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+     the JSON `Map` into a `Album` using the `fromJson()` factory method.
+  3. If the server does not return an CREATED response with a status code of 201,
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: does not return => doesn't return
an CREATED => a `CREATED`

File: src/docs/cookbook/networking/send-data.md [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
+    // If the server did return a 201 CREATED response, then parse the JSON.
+    return Album.fromJson(json.decode(response.body));
+  } else {
+    // If the server did not return a 201 CREATED response, then throw an exception.
+    throw Exception('Failed to load album');
+  }
+}
+```
+
+Hooray! Now you've got a function that send the title to a server to create an album.
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: that send the => that sends the

File: src/docs/cookbook/networking/send-data.md [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+
+## 4. Let's get the title as input from user
+
+Let's create a `TextField` to enter title and a `RaisedButton` to send data to server.
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: to enter title => to enter a title

File: src/docs/cookbook/networking/send-data.md [hunk 6]
```diff
@@ -0,0 +1,334 @@
...
+  }
+}
+```
+
+Hooray! Now you've got a function that send the title to a server to create an album.
+
+## 4. Let's get the title as input from user
+
+Let's create a `TextField` to enter title and a `RaisedButton` to send data to server. 
+Also define a `TextEditingController` to read the user input form a `TextField`
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: input form a => input from a

File: src/docs/cookbook/networking/send-data.md [hunk 7]
```diff
@@ -0,0 +1,334 @@
...
+        setState(() {
+          _futureAlbum = createAlbum(_controller.text);
+        });
+      },
+    ),
+  ],
+)
+```
+
+On pressing the button `Create Data`, we make the network request,
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: `Create Data` => **Create Data**

File: src/docs/cookbook/networking/send-data.md [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+which will send the data in the `TextField` to the server with a `POST` request.
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: which will send => which sends

File: src/docs/cookbook/networking/send-data.md [hunk 9]
```diff
@@ -0,0 +1,334 @@
...
+The `FutureBuilder` widget comes with Flutter and
+makes it easy to work with async data sources.
+You must provide two parameters:
+
+  1. The `Future` you want to work with. In this case, the future returned from
+  the `createAlbum()` function.
+  2. A `builder` function that tells Flutter what to render, depending on the
+  state of the `Future`: loading, success, or error.
+
+Note that `snapshot.hasData` will only return `true` when the snapshot contains
```
Comment (sfshaza2) [inline src/docs/cookbook/networking/send-data.md]: will only return => only returns

**Comment by hemanthrajv** [comment]
@sfshaza2 , [this](https://github.com/hemanthrajv/website/commit/d0680385446a15553a8bbae51959b06a72f00f36) commit addresses all above requested changes as per guidelines.

**Pull request merged**